src/main/kotlin/me/zeroeightsix/kami/feature/module/Nuker.kt

```diff
@@ -0,0 +1,128 @@
+            val block = currentBlock ?: return@Listener
+            val state = it.world.getBlockState(block)
+
+            if (selectTool)
+                AutoTool.equipBestTool(state)
+
+            if (progress == 0.0)
+                mine(block, true)
+
+
+            progress += state.calcBlockBreakingDelta(it.player, it.world, block)
+            if (progress >= 1.0) {
+                mine(block, false)
+                progress = 0.0
+                currentBlock = null
+            }
+        }
+    })
+
+    private fun getBoxCorner(playerPos: Vec3d, negative: Boolean = false) =
+        playerPos.add(Vec3d(hitRange, hitRange, hitRange).run { if (negative) negate() else this })
+
+    /**
+     * mines a block using packets
+     * @param op if true, mining is started, if false it is stopped
+     */
+    private fun mine(block: BlockPos, op: Boolean) {
+        mc.networkHandler?.sendPacket(
+            PlayerActionC2SPacket(
+                if (op) PlayerActionC2SPacket.Action.START_DESTROY_BLOCK
+                else PlayerActionC2SPacket.Action.STOP_DESTROY_BLOCK,
+                block,
+                Direction.DOWN
+            )
+        )
+    }
+
+    private fun nextBlock(player: ClientPlayerEntity, world: ClientWorld) =
+        BlockPos.method_29715(Box(getBoxCorner(player.pos), getBoxCorner(player.pos, true)))
+            .filter { validate(player, world, it) }.findFirst().kotlin
+
+    private fun instantMineBlocks(player: ClientPlayerEntity, world: ClientWorld) {
+        BlockPos.method_29715(Box(getBoxCorner(player.pos), getBoxCorner(player.pos, true)))
+            .filter { validate(player, world, it) && canInstaMine(player, world, it) }
+            .forEach { mine(it, true) }
+    }
+
+    private fun canInstaMine(player: ClientPlayerEntity, world: ClientWorld, block: BlockPos) =
+        player.isCreative || world.getBlockState(block).calcBlockBreakingDelta(player, world, block) >= 1
+
+    private fun validate(player: ClientPlayerEntity, world: ClientWorld, block: BlockPos): Boolean {
+        val state = world.getBlockState(block)
+
+        val throughBlockCheck = {
+            throughBlocks ||
+                world.raycast(
+                    RaycastContext(
+                        Vec3d(player.pos.x, player.eyeY, player.pos.z),
+                        block.asVec3d.add(singleVec(0.5)),
+                        RaycastContext.ShapeType.COLLIDER,
+                        RaycastContext.FluidHandling.NONE,
+                        player
+                    )
+                ).blockPos == block
+        }
+
+        val blockCheck = {
+            !state.isAir &&
+                state.block !is FluidBlock &&
+                block.isWithinDistance(
+                    player.pos,
+                    hitRange
+                ) &&
+                (player.isCreative || state.getHardness(world, block) >= 0)
+        }
+
+        return blockCheck() && throughBlockCheck()
```

Comment:
Err, sure, but that's not very pretty..

Just add a `if (!throughBlockCheck) return false` in between `throughBlocksCheck` and `blockCheck` instead of this. (Which would also simplify your last return to `return blockCheck`, as `throughBlocksCheck` is always true by the time you'd reach the last return.)